### DISCUSSION TITLE
Change state of column when exporting

### DISCUSSION THREAD

**Discussion by stblink** [filamentphp/filament#13435] [Help]
### Package

Table builder

### Package Version

v3.2.92

### How can we help you?

My InterventionsResource file generates a table column and replaces the value from 1 to YES and 0 to NO according to this code:

`Tables\Columns\TextColumn::make('on_site')
                    ->formatStateUsing(fn ($record) => $record->on_site == 1 ? 'Yes' : 'No') 
                    ->toggleable(isToggledHiddenByDefault: true)`

This works perfectly fine in the resource file, but not in the export class.
I want to export this table and following the docs i added to the Bulk actions the following code:

`ExportBulkAction::make()
                        ->exporter(InterventionsExporter::class),
                ])`

Having already done the migrations, notifications and all that its needed. The excel file is generated but the column values are not formatted using the `formatStateUsing` method. So in InterventionExporter class i have the following code which exports without any format:

`ExportColumn::make('on_site')
            ->formatStateUsing(fn ($record) => $record->on_site == 1 ? 'Yes' : 'No')`

This will output the same result as if `formatStateUsing` wasn't used at all (exported xlsx file shows 1 or 0). But for example `->label('column heading')` works perfectly.
Tried using a different exporter field column using the same method and it doesnt work either. For example i have a field which is start_date and inside the export class:

```
ExportColumn::make('start_date')
->formatStateUsing(fn (string $state) => Carbon::parse($state)->format('M d, Y'))
```

will output always unformatted date:
![image](https://github.com/filamentphp/filament/assets/44067961/a4691cf9-7f4c-4fbe-ae8d-c134682233d0)

Checked and re-checked the docs, searched everywhere and still can't find the problem. Any help would be most apprecciated.

Here is the exporter class file content:

```
namespace App\Filament\Exports;

use App\Models\Interventions;
use Filament\Actions\Exports\ExportColumn;
use Filament\Actions\Exports\Exporter;
use Filament\Actions\Exports\Models\Export;
use Illuminate\Support\Carbon;

class InterventionsExporter extends Exporter
{
    protected static ?string $model = Interventions::class;

    public static function getColumns(): array
    {
        return [
            ExportColumn::make('id')
                ->label('ID'),
            ExportColumn::make('client.name'),
            ExportColumn::make('description'),
            ExportColumn::make('notes'), 
            ExportColumn::make('start_date')
            ->formatStateUsing(fn (string $state) => Carbon::parse($state)->format('M d, Y')),
            ExportColumn::make('on_site')
            ->formatStateUsing(fn ($record) => $record->on_site == 1 ? 'Yes' : 'No'),
            ExportColumn::make('start_time'),
            ExportColumn::make('end_time'),
            ExportColumn::make('duration'),
            ExportColumn::make('requested_by'),
            ExportColumn::make('created_at'),
            ExportColumn::make('updated_at'),
        ];
    }

    public static function getCompletedNotificationBody(Export $export): string
    {
        $body = 'Your interventions export has completed and ' . number_format($export->successful_rows) . ' ' . str('row')->plural($export->successful_rows) . ' exported.';

        if ($failedRowsCount = $export->getFailedRowsCount()) {
            $body .= ' ' . number_format($failedRowsCount) . ' ' . str('row')->plural($failedRowsCount) . ' failed to export.';
        }

        return $body;
    }
}
```

**Comment by simonpeters** [comment]
I'm having the same issue. the formatStateUsing doesn't seem to do anything on the ExportColumn even though the docs says it should work

**Reply by muzhawir** [reply]
I have the same problem, so I want the 'gender' column to be exported with the changed cell value like this:

```php
  ExportColumn::make('gender')
      ->formatStateUsing(fn ($record) => $record->gender === 'M' ? 'Laki-Laki' : 'Perempuan'),
```

but the exported output does not change:
![image](https://github.com/filamentphp/filament/assets/7391703/af2ec02c-b60c-42f8-81ce-7f994d1ba138)

**Reply by muzhawir** [reply]
Solved! I just update my filament version and restart my queue.

![image](https://github.com/filamentphp/filament/assets/7391703/3e107440-e659-4c7a-bd57-200a8041a8d2)

@stblink @simonpeters Try it!

**Comment by stblink** [comment]
Thanks @muzhawir , i ended up going or Laravel/Excel. Will give it a look in future projects, glad it worked out for you!

**Comment by Kreyu** [comment]
I encountered this issue using Filament v3.3.3. Turns out this was an issue on my side, as I was running `queue:work` locally in Docker container with supervisor, so I had to restart the queue after each change.

**Reply by ramees-intersmart** [reply]
Thanks, restarting queue worked for me
